src/indra_cogex/client/neo4j_client.py

```diff
@@ -77,6 +77,21 @@ def __del__(self):
         if self.driver is not None:
             self.driver.close()
 
+    def load_agent_cache(self):
+        """Load all agents into a dictionary cache for quick lookup."""
+        query = "MATCH (n:BioEntity) RETURN n.id AS id, n.name AS name"
```

Comment:
We should add a constraint that the BioEntity is connected (arbitrary direction) to an :indra_rel.